powerauth-java-server/src/test/java/com/wultra/powerauth/fido2/Fido2AuthenticatorTest.java

```diff
@@ -0,0 +1,440 @@
+        final PublicKeyCredential<AuthenticatorAttestationResponse, RegistrationExtensionClientOutput> credential = CLIENT_PLATFORM_SELF_ATTESTED.create(credentialCreationOptions);
+        com.wultra.powerauth.fido2.rest.model.request.RegistrationRequest registrationRequest = new RegistrationRequest();
+        registrationRequest.setApplicationId(APPLICATION_ID);
+        registrationRequest.setActivationName(ACTIVATION_NAME);
+        registrationRequest.setExpectedChallenge(new String(challenge.getValue(), StandardCharsets.UTF_8));
+        final AuthenticatorParameters authenticationParameters = new AuthenticatorParameters();
+        authenticationParameters.setId(credential.getId());
+        authenticationParameters.setRelyingPartyId(RP_ID);
+        authenticationParameters.setAllowedOrigins(Collections.singletonList(ORIGIN.toString()));
+        authenticationParameters.setType(credential.getType());
+        authenticationParameters.setRequiresUserVerification(true);
+
+        // Convert clientDataJSON and attestationObject into object and supply encoded values for signature verification
+        final byte[] clientDataJSON = Objects.requireNonNull(credential.getAuthenticatorResponse()).getClientDataJSON();
+        final CollectedClientData clientData = OBJECT_MAPPER.readValue(clientDataJSON, CollectedClientData.class);
+        clientData.setEncoded(new String(clientDataJSON));
+        final byte[] attestationObject = Objects.requireNonNull(credential.getAuthenticatorResponse()).getAttestationObject();
+        final AttestationObject attObj = CBOR_MAPPER.readValue(attestationObject, AttestationObject.class);
+        attObj.setEncoded(new String(attestationObject));
+
+        final com.wultra.powerauth.fido2.rest.model.entity.AuthenticatorAttestationResponse attestationResponse = new com.wultra.powerauth.fido2.rest.model.entity.AuthenticatorAttestationResponse();
+        attestationResponse.setClientDataJSON(clientData);
+        attestationResponse.setAttestationObject(attObj);
+        final AuthenticatorTransport[] transports = credential.getAuthenticatorResponse().getTransports().toArray(new AuthenticatorTransport[0]);
+        attestationResponse.setTransports(Arrays.stream(transports).map(AuthenticatorTransport::toString).collect(Collectors.toList()));
+        authenticationParameters.setResponse(attestationResponse);
+        registrationRequest.setAuthenticatorParameters(authenticationParameters);
+        return registrationRequest;
+    }
+
+    private void createOperationTemplate() throws Exception {
+        final OperationTemplateListResponse templateList = powerAuthService.getAllTemplates();
+        boolean templateFound = false;
+        for (OperationTemplateDetailResponse template: templateList) {
+            if ("login".equals(template.getTemplateName())) {
+                templateFound = true;
+                break;
+            }
+        }
+        if (templateFound) {
+            return;
+        }
```

Comment:
May be written in Java 8 style in this way:
```suggestion
        final boolean templateFound = powerAuthService.getAllTemplates().stream()
                .map(OperationTemplateDetailResponse::getTemplateName)
                .anyMatch("login"::equals);
        if (templateFound) {
            return;
        }
```